include/zephyr/net/dns_resolve.h

```diff
@@ -526,12 +547,14 @@ int dns_resolve_close(struct dns_resolve_context *ctx);
  * @param servers_sa DNS server addresses as struct sockaddr. The array
  * is NULL terminated. Port numbers are optional in struct sockaddr, the
  * default will be used if set to 0.
+ * @param source Source of the DNS servers, e.g., manual, DHCPv4/6, etc.
  *
  * @return 0 if ok, <0 if error.
  */
 int dns_resolve_reconfigure(struct dns_resolve_context *ctx,
 			    const char *servers_str[],
-			    const struct sockaddr *servers_sa[]);
+			    const struct sockaddr *servers_sa[],
+			    enum dns_server_source source);
```

Comment:
Probably worth to mention this change in the migration guide. The API is unstable so we don't need to go any further I think?

Reply:
Sure, I will add that